appIcons.js

```diff
@@ -316,8 +316,9 @@ class MyAppIcon extends Dash.DashIcon {
                     let fixedDock = Docking.DockManager.settings.get_boolean('dock-fixed');
                     let additional_margin = this._isHorizontal && !fixedDock ? Main.overview.dash.height : 0;
                     let verticalMargins = this._menu.actor.margin_top + this._menu.actor.margin_bottom;
+                    let scale = St.ThemeContext.get_for_stage(global.stage).scale_factor;
                     // Also set a max width to the menu, so long labels (long windows title) get truncated
-                    this._menu.actor.style = ('max-height: ' + Math.round(workArea.height - additional_margin - verticalMargins) + 'px;' +
+                    this._menu.actor.style = ('max-height: ' + parseInt((workArea.height - additional_margin - verticalMargins)/scale,10) + 'px;' +
```

Comment:
Is there a reason to use `parseInt` here (and elsewhere) over `Math.floor`? `parseInt` is slightly more complex in that it is typically used to parse strings into integers, with the radix as optional/kind-of-mandatory 2nd param.

Reply:
You can use several different methods to convert to a number, however based on all the speed tests I've seen the difference between them all is minuscule, with `parseInt` a tad faster than `math.round`, but slower than say (+"string_number").   However, parseInt in my opinion is more descriptive than Math.round and using (+"string") and so I tend to use it for number conversion where I am clipping the number just because it is so self documenting that we are clipping the number, not rounding it in any direction.   :grinning: